### PR TITLE
[Typing][PEP585 Upgrade][BUAA][41-50] Use standard collections for type hints for 10 files in `python/paddle/distribution`

### PR DESCRIPTION
**Pull request by Caogration**
### PR Category

User Experience

### PR Types

Devs

### Description

为如下文件添加 PEP 563，并进行 PEP 585 升级：

- python/paddle/distribution/exponential.py
- python/paddle/distribution/gamma.py
- python/paddle/distribution/geometric.py
- python/paddle/distribution/gumbel.py
- python/paddle/distribution/independent.py
- python/paddle/distribution/laplace.py
- python/paddle/distribution/lognormal.py
- python/paddle/distribution/normal.py
- python/paddle/distribution/transform.py
- python/paddle/distribution/transformed_distribution.py

### Related links

- #66936

@gouzil

### CODE REVIEW

File: python/paddle/distribution/normal.py [hunk 1]
```diff
@@ -14,8 +14,8 @@
 from __future__ import annotations
 
 import math
-from collections.abc import Iterable
-from typing import TYPE_CHECKING, Sequence, Union
+from collections.abc import Iterable, Sequence
+from typing import TYPE_CHECKING, Union
```
Comment (gouzil) [inline python/paddle/distribution/normal.py]: 把`Union` 移到 `TYPE_CHECKING` 导入

Comment (Caogration) [reply python/paddle/distribution/normal.py]: Done

File: python/paddle/distribution/lognormal.py [hunk 2]
```diff
@@ -13,14 +13,16 @@
 # limitations under the License.
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Sequence, Union
+from typing import TYPE_CHECKING, Union
```
Comment (gouzil) [inline python/paddle/distribution/lognormal.py]: 同上

Comment (Caogration) [reply python/paddle/distribution/lognormal.py]: Done

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**